core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -242,6 +253,7 @@ public static TableMetadata read(FileIO io, InputFile file) {
     }
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
should probably refactor the method into methods like `parsePartitionSpecs`, `parseSortOrders` and `parsePrimaryKeys`, but it is not urgent, we can do it in another PR.